050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,680 @@
+  2. Deleting the `KafkaTopic` from the old namespace,
+  3. Revising the policy (requiring a UTO restart)
+  4. Creating a new `KafkaTopic` in the new namespace.
+
+#### Changing `spec.topicName`
+
+This will NOT be supported.
+It will be detected by a mismatch between `spec.topicName` and `status.topicName`.
+Changes thus detected will result in an error condition.
+
+#### Decreasing `spec.partitions`
+
+This will NOT be supported, because it's not supported by Kafka.
+The user will either need to revert the `spec.partitions`, or recreate the topic in Kafka (which they can do via the `KafkaTopic` or not).
+
+#### Changing `spec.relicas`
+
+This will NOT be supported using as part of this proposal.
+
+#### Recreating topics in the Kafka cluster
+
+This WILL be supported.
+
+#### Deployment of application with `KafkaTopics`
+
+Consider an application deployed via a manifest containing a `Deployment` and some `KafkaTopics`.
+If the Kafka cluster is configured with `auto.create.topics.enable` there is a race condition between:
+
+1. the operator reacting to the new `KafkaTopics`.
+2. the application starting and triggering autocreation of the topics.
+
+**If the operator wins:** then auto-creation will not happen, and the user's intent is honoured.
+
+**If the application wins:** then the topics will be created using the default configuration for the cluster. The operator will later reconcile the `KafkaTopics` which will result in reconfiguration the default configuration for auto-created topics differs from the `KafkaTopic.spec`. That reconfiguration will either succeed, or fail (e.g. because a `KafkaTopic` has a lower `spec.partitions` than the auto-created topic).
+
+To avoid this it is **recommended** that the Topic Operator not be used with cluster where `auto.create.topics.enable` is `true`. 
+The operator will emit a `WARN` level log on startup if autocreation is enabled. 
+
+Alternatively: App could be written to wait for topic existency, or use an init container to do the same.
+
+## Affected/not affected projects
+
+* The existing bidirectional Topic Operator would be deprecated and removed once ZooKeeper-based clusters are no longer supported.
+* A new unidirectional Topic Operator would be provided as a replacement, which users could migrate to before ZooKeeper-based clusters are no longer supported.
+* The schema of `KafkaTopic` would change
+* The schema of `Kafka` would also change, to support deployment of the new operator, but the details of that are not defined in thisproposal.
+* The CRD changes would be relfected in the `api` module.
+
+## Compatibility
+
+// Call out any future or backwards compatibility considerations this proposal has accounted for.
+
+To migrate, existing BTO users would have to:
+
+1. Review how they're using the BTO and whether they require birdirectionality
+2. Undeploy the BTO (they can retain their existing `KafkaTopics`)
+3. Deploy the UTO
+
+
+### Disadvantages
```

Comment:
The use of finalizers should be listed in the disadvantages as well I think. It is something that might have its justification, but it also brings issues to deletions.

Reply:
Yes, done.